go.mod

```diff
@@ -1,8 +1,6 @@
 module github.com/thanos-io/thanos
 
-go 1.22.0
-
-toolchain go1.22.5
+go 1.23.0
```

Comment:
should we maybe bundle this on a separate PR? This interferes with build, maybe we should bump to 1.23 all in one pass.

Reply:
Technically we don't need to bump the version in `go.mod` because that indicates the minimum required version. If we do a bump without using new features, it means that downstream users are also forced to update their version even though there's no reason for that.

https://go.dev/doc/modules/gomod-ref#go